ashlar/views.py

```diff
@@ -71,6 +73,16 @@ class BoundaryViewSet(viewsets.ModelViewSet):
     serializer_class = BoundarySerializer
     filter_class = BoundaryFilter
 
+    def create(self, request, *args, **kwargs):
+        try:
+            super(BoundaryViewSet, self).create(request, *args, **kwargs)
+        except IntegrityError:
+            return Response({'error': 'uniqueness constraint violation'}, status.HTTP_409_CONFLICT)
+
+    def perform_create(self, serializer):
```

Comment:
For these file uploads; creation requires only label and source_file - modifying perform_create allows us to create our instance with the uploaded file. 500s abound otherwise.